app/mailers/notification_mailers/mentioned.rb

```diff
@@ -4,9 +4,10 @@ class Mentioned < NotificationMailers::Base
     delegate :author_name, to: :post, prefix: true
 
     def set_headers(target_id)
-      @post = Mention.find_by_id(target_id).post
+      @post = Mention.find_by_id(target_id).mentions_container
 
       @headers[:subject] = I18n.t('notifier.mentioned.subject', :name => @sender.name)
+      @headers[:template_name] = "mentioned_in_comment" if post.is_a?(Comment)
```

Comment:
However I'm not sure about `Workers::Mail::Base` as a good name. First it'll make confusion with `Workers::Base`. Second not every `Workers::Mail::*` will be child of this `Workers::Mail::Base`. There are `ReportWorker` or `InviteEmail` who have different logic. All `Workers::Mail::*` that we can commonify use `Notifier` class. So we can call the base class for it `Workers::Mail::NotifierBase` or something. ?